extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -82,15 +82,19 @@ private[kinesis] class KinesisReceiver(
   var workerId: String = null
 
   /*
-   * This impl uses the DefaultAWSCredentialsProviderChain and searches for credentials 
-   *   in the following order of precedence:
+   * This impl uses the DefaultAWSCredentialsProviderChain unless it's provided by constructor
+   *  and searches for credentials in the following order of precedence:
    * Environment Variables - AWS_ACCESS_KEY_ID and AWS_SECRET_KEY
    * Java System Properties - aws.accessKeyId and aws.secretKey
    * Credential profiles file at the default location (~/.aws/credentials) shared by all 
    *   AWS SDKs and the AWS CLI
    * Instance profile credentials delivered through the Amazon EC2 metadata service
    */
-  var credentialsProvider: AWSCredentialsProvider = null
+  var credentialsProvider: AWSCredentialsProvider = (credentials map { cr => new AWSCredentialsProvider {
```

Comment:
i don't know if there's a simpler way but i'll think about it.

although the unit tests are passing now, i'll run example applications to check if the serialization issue you mentioned still exists